### PR TITLE
Alex add jsd device base and generic sdo cmds

### PR DESCRIPTION
**Pull request by alex-brinkman**
Adds interfaces for JSD device SDO commands
* Adds `async_sdo_write` and `async_sdo_read` DeviceCmds
* Introduces new `JsdDeviceBase` base class for online/offline JSD devices
  * add default online and offline SDO requests handling
  * Adds UTs for offline SDO requests

### CODE REVIEW

**Comment by alex-brinkman** [comment]
@d-loret merging without your review comments for the reasons we discussed (e.g. delivery schedules). If you want to review and comment on this PR I'll commit to responding to every comment.

File: src/manager.cc [hunk 1]
```diff
@@ -272,6 +277,23 @@ bool fastcat::Manager::Process()
...
+  // for each JSD context, pop their queues and push them onto the single 
+  // fastcat queue
+  SdoResponse entry;
+  for (auto it = jsd_map_.begin(); it != jsd_map_.end(); ++it) {
+    while(jsd_sdo_pop_response_queue(it->second, &entry.response)){
+      
+      if(entry.response.slave_id < *(it->second->ecx_context.slavecount) ){
+        entry.device_name = it->second->slave_configs[entry.response.slave_id].name;
+      }else{
+          entry.device_name = "invalid name";
```
Comment (d-loret) [inline src/manager.cc:289]: @alex-brinkman, I think `entry.response.success` should be set to `false` here. When processing the responses, it is better to rely on the boolean flag to determine whether the response is valid.

File: test/test_unit/test_jsd_device_base.cc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+
+TEST_F(JsdDeviceBaseTest, RegisterAndUseSdo){
+
+  device_.RegisterSdoResponseQueue(sdo_response_queue_);
+
+  fastcat::DeviceCmd cmd;
+
+  cmd.name                             = "patron";
```
Comment (d-loret) [inline test/test_unit/test_jsd_device_base.cc:50]: I don't know about this... I would use Don Julio.

File: src/jsd/jsd_device_base.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  AsyncSdoRetVal WriteAsyncSdoRequestOnline(DeviceCmd& cmd);
+  AsyncSdoRetVal WriteAsyncSdoRequestOffline(DeviceCmd& cmd);
+
+  bool   is_offline_ = false; ///< If is an offline version
+  jsd_t* context_ = NULL; ///< JSD context
+  int    slave_id_ = 0;   ///< EtherCAT Slave Index
+
+  /// provided so that offline devices can push to this queue with 
+  /// SDO response stubs
+  std::shared_ptr<std::queue<SdoResponse>> sdo_response_queue_;
```
Comment (d-loret) [inline src/jsd/jsd_device_base.h:49]: Is this only intended for offline devices? I saw online devices register the queue.

In order to process the SDO responses, is the application supposed to call the manager's `PopSdoResponseQueue`?

**Pull request merged**